workflow.py

```diff
@@ -214,67 +228,107 @@ async def eliminate_subtraction(
     """
+
     to_subtraction_sam_path = work_path / "to_subtraction.sam"
 
-    # Map reads to the subtraction.
-    await run_subprocess(
-        [
-            "bowtie2",
-            "--local",
-            "--no-unal",
-            "--no-hd",
-            "--no-sq",
-            "-N",
-            "0",
-            "-p",
-            str(proc),
-            "-x",
-            shlex.quote(str(subtraction.bowtie2_index_path)),
-            "-U",
-            str(isolate_fastq_path),
-            "-S",
-            str(to_subtraction_sam_path),
-        ]
-    )
+    current_fastq_path = work_path / "current_fastq.fq"
+    new_fastq_path = work_path / "new_fastq.fq"
 
-    await run_subprocess(
-        [
-            "./eliminate_subtraction",
-            str(isolate_sam_path),
-            str(to_subtraction_sam_path),
-            str(subtracted_sam_path),
-        ]
-    )
+    # copy the original fastq file into a working fastq file
+    # as to not disrupt possible uses elsewhere
+    await asyncio.to_thread(shutil.copyfile, isolate_fastq_path, current_fastq_path)
 
-    await aiofiles.os.remove(to_subtraction_sam_path)
+    # for the first subtraction, use the original isolate file
+    current_isolate_path = isolate_sam_path
 
     subtracted_count = 0
 
-    async with aiofiles.open("subtracted_read_ids.txt", "r") as f:
-        async for line in f:
-            if line:
-                subtracted_count += 1
-
-    logger.info(
-        f"Subtracted {subtracted_count} reads that mapped better to a subtraction."
-    )
+    # iterate over all subtractions
+    for subtraction in subtractions:
+        # Map reads to the subtraction.
+        await run_subprocess(
+            [
+                "bowtie2",
+                "--local",
+                "--no-unal",
+                "--no-hd",
+                "--no-sq",
+                "-N",
+                "0",
+                "-p",
+                str(proc),
+                "-x",
+                shlex.quote(str(subtraction.bowtie2_index_path)),
+                "-U",
+                str(current_fastq_path),
+                "-S",
+                str(to_subtraction_sam_path),
+            ]
+        )
+
+        await run_subprocess(
+            [
+                "./eliminate_subtraction",
+                str(current_isolate_path),
+                str(to_subtraction_sam_path),
+                str(subtracted_sam_path),
+            ]
+        )
+
+        await aiofiles.os.remove(to_subtraction_sam_path)
+
+        # set next iteration's current isolate file to the newly created isolate file
+        current_isolate_path = work_path / "working_isolate.sam"
+        await asyncio.to_thread(
+            shutil.copyfile,
+            subtracted_sam_path,
+            current_isolate_path
+        )
+
+        async with aiofiles.open("subtracted_read_ids.txt", "r") as f:
+            async for line in f:
+                if line:
+                    subtracted_count += 1
```

Comment:
You open and read this file again below. What about the following:
```py
async with aiofiles.open("subtracted_read_ids.txt", "r") as f:
    subtracted_read_ids = {line.strip("@\n") for line in subtracted_reads_file if line}

subtracted_count  = len(subtracted_read_ids)
```
You can then use the `set` when you are filtering the FASTQ records below.